File: Python/Lab_2/exhibits/Shield.py
from exhibits.Exhibit import *
from enums.ExhibitName import *
from enums.ExhibitAge import *


class Shield(Exhibit):
    exhibit_age = ExhibitAge.SEVEN_BC
    exhibit_name = ExhibitName.SHIELD

    def __init__(self, shape, name, age, kind_of_art, place_of_manufacture, condition, weight):
        self.shape = shape
        Exhibit.__init__(self, name, age, kind_of_art, place_of_manufacture, condition, weight)

    def __str__(self):
        return "Name : " + str(self.name) + " age: " + str(self.age) + " the shape of shield is " + str(
            self.shape)

File: Lab_3/Shield.java
package ua.lviv.iot;

public class Shield extends Exhibit{
    private String shape;
    
    public Shield() {
    	
    }
     public Shield(ExhibitName name, ExhibitAge age, String kindOfArt, String placeOfManufacture, String condition, double weight, String shape) {
    	     super(name, age, kindOfArt, placeOfManufacture, condition, weight);
    	     this.shape = shape;
     }
	public String getShape() {
		return shape;
	}
	public void setShape(String shape) {
		this.shape = shape;
	}
	
	@Override
	public String toString() {
		return super.toString() + ", shape: " + this.shape + "\n";
	}
}
